repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -1019,8 +1019,13 @@ private AtlasEntityHeader mapVertexToAtlasEntityHeader(AtlasVertex entityVertex,
             ret.setTypeName(typeName);
             ret.setGuid(guid);
             ret.setStatus(GraphHelper.getStatus(entityVertex));
-            if(RequestContext.get().includeClassifications()){
-                ret.setClassificationNames(getAllTraitNames(entityVertex));
+            RequestContext context = RequestContext.get();
+            boolean includeClassifications = context.includeClassifications();
+            boolean includeClassificationNames = context.isIncludeClassificationNames();
+            if(includeClassifications){
+                ret.setClassificationNames(getAllTraitNamesFromAttribute(entityVertex));
+            } else if (!includeClassifications && includeClassificationNames) {
```

Comment:
nit: `&& includeClassificationNames` is redundant in `else if`.

Reply:
It is not redundant actually, this has a purpose;
This is making sure that `includeClassificationNames` gets applied if and only if `includeClassifications` set as false

Reply:
Yes, my bad for the typo, I meant `!includeClassifications` is redundant due to the `else`.

Reply:
Oh Yes!

Reply:
Ignoring this for now